stdlib/Test/src/logging.jl

```diff
@@ -29,6 +42,40 @@ mutable struct TestLogger <: AbstractLogger
     message_limits::Dict{Any,Int}
 end
 
+"""
+    TestLogger(; min_level=Info, catch_exceptions=false)
+
+Create a `TestLogger` which captures logged messages in its `logs::Vector{LogRecord}` field.
+
+See also: [`LogRecord`](@ref).
+
+## Example
+
+```jldoctest
+julia> using Test, Logging
+
+julia> f() = @info "Hi" number=5
+f (generic function with 1 method)
```

Comment:
To save some vertical space:
```suggestion
julia> f() = @info "Hi" number=5;
```